Firestore/Source/API/FIRVectorValue.mm

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#include <vector>
+
+#include "Firestore/Source/Public/FirebaseFirestore/FIRVectorValue.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation FIRVectorValue
+
+@synthesize array = _internalValue;
+
+- (instancetype)initWithArray:(NSArray<NSNumber *> *)array {
+  if (self = [super init]) {
+    _internalValue = [NSArray arrayWithArray:array];
```

Comment:
```suggestion
    _array = [NSArray arrayWithArray:array];
```

Reply:
This LGTM. The one small caveat here now is that an Objective-C client calling `vector.array` will get `NSArray<NSNumber>` back whose numbers have not been truncated. I think this is okay, but just calling it out in case it's not.